### DISCUSSION TITLE
Change syntax highlighting

### DISCUSSION THREAD

**Discussion by hajonnes** [nvim-treesitter/nvim-treesitter#5726] [General]
Hi,
I use comments as subtitles in a program and I would like to change the highlighting of the comments so they stand out better.  
When I put the cursor over a highlight and do the ex command `:Inspect` I get:  
```
Treesitter                                                                                                                                                 
  - @comment.c links to Comment c
  - @spell.c links to @spell c
 ```
From this information, is it possible to change highlighting?

**Comment by clason** [comment]
Yes, just `:highlight @comment` however you want.

**Reply by hajonnes** [reply]
Sorry, I do not understand how I should write it.
Could you give me an example, e.g. if I want the comments to be "red"
Also is there a way to permanently set this value in init.lua?

**Reply by clason** [reply]
Yes, of course, but this is basic information you can find in the documentation; please read that first: `:h treesitter-highlight-groups`.

**Reply by hajonnes** [reply]
To be honest, I do not understand the docs. Like:  
```
Syntax highlighting is specified through queries named `highlights.scm`,
which match a `TSNode` in the parsed `TSTree` to a `capture` that can be
assigned a highlight group. 
```
Does this mean that the nodes have a query per node and the tree is traversed and if a match is found in the tree that node has a  return value called a "capture"? 

```
For example, the query

    (parameters (identifier) @parameter)

matches any `identifier` node inside a function `parameter` node (e.g., the
`bar` in `foo(bar)`) to the capture named `@parameter`.
```
So an identifier can be any name of a variable or function.
What is an identifier node?
What is a function parameter node?
Perhaps I need to read the whole manual to understand the text. At least for me I do have a hard time understanding.  I got some good examples that I could see the syntax from, like `hi @comment.c guifg=Blue`  though.
If we take  `hi @comment.c guifg=Blue` as an example, how can I permanently put this option in init.lua?

**Reply by clason** [reply]
Yes, you need to read the complete docs. Tree-sitter is an advanced topic and not for the faint of heart currently. Maybe ignore this for now while you're learning basic Neovim.

> If we take  hi @comment.c guifg=Blue as an example, how can I permanently put this option in init.lua?

Read `:h lua-guide`.
